src/game/engine/crate.cpp

```diff
@@ -88,14 +90,17 @@ BOOL CrateClass::Put_Crate(cell_t &cell)
 
 BOOL CrateClass::Get_Crate(cell_t cell)
 {
-    // TODO Requires MapClass, CellClass and ScenarioClass to actuall implement the functions.
-#ifdef GAME_DLL
-    BOOL (*func)(cell_t) = reinterpret_cast<BOOL (*)(cell_t)>(0x004ACB1C);
-    return func(cell);
-#else
-    DEBUG_ASSERT_PRINT(false, "Unimplemented function called!\n");
+    if (Map.In_Radar(cell)) {
+        CellClass *cptr = &Map[cell];
```

Comment:
Just get a `CellClass &` here instead of a pointer and use the reference syntax of '.' rather than '->' . If there is no pointer arithmetic and no null check, then its best to use reference in C++.